src/Deserializers/EntityDeserializer.php

```diff
@@ -113,24 +114,46 @@ private function setAliasesFromSerialization( array $serialization, Entity $enti
 		}
 		$this->assertAttributeIsArray( $serialization, 'aliases' );
 
-		foreach( $serialization['aliases'] as $aliasesPerLanguageSerialization ) {
+		foreach( $serialization['aliases'] as $requestedLanguage => $aliasesPerLanguageSerialization ) {
 			if ( !is_array( $aliasesPerLanguageSerialization ) ) {
 				throw new DeserializationException( "Aliases attribute should be an array of array" );
 			}
 
 			foreach( $aliasesPerLanguageSerialization as $aliasSerialization ) {
-				$this->assertIsValidValueSerialization( $aliasSerialization );
+				$this->assertIsValidValueSerialization( $aliasSerialization, $requestedLanguage );
 				$entity->addAliases( $aliasSerialization['language'], array( $aliasSerialization['value'] ) );
 			}
 		}
 	}
 
-	private function assertIsValidValueSerialization( $serialization ) {
+	private function assertNotAttribute( array $array, $key ) {
+		if ( array_key_exists( $key, $array ) ) {
+			throw new InvalidAttributeException(
+				$key,
+				$array[$key],
+				'Deserialization of attribute '.$key.' not supported.'
+			);
+		}
+	}
+
+	private function assertRequestedAndActualLanguageMatch( $serialization, $requestedLanguage ) {
+		if ( $serialization['language'] !== $requestedLanguage ) {
+			throw new DeserializationException(
+				'Deserialization of a value of the attribute language (actual)'
+					.' that is not matching the language key (requested) is not supported: '
+					.$serialization['language'].' !== '.$requestedLanguage
+			);
+		}
+	}
+
+	private function assertIsValidValueSerialization( $serialization, $requestedLanguage ) {
 		$this->requireAttribute( $serialization, 'language' );
 		$this->requireAttribute( $serialization, 'value' );
+		$this->assertNotAttribute( $serialization, 'source' );
 
 		$this->assertAttributeInternalType( $serialization, 'language', 'string' );
 		$this->assertAttributeInternalType( $serialization, 'value', 'string' );
+		$this->assertRequestedAndActualLanguageMatch( $serialization, $requestedLanguage );
```

Comment:
I'm not a fan of code that's split up in so many trivial methods. But this patch does not introduces this coding style, so no reason to block it.